datadog_lambda/tracing.py

```diff
@@ -660,18 +623,32 @@ def is_lambda_context():
 
 def set_dd_trace_py_root(trace_context_source, merge_xray_traces):
     if trace_context_source == TraceContextSource.EVENT or merge_xray_traces:
-        context = dict(dd_trace_context)
+        global dd_trace_context
+            
         if merge_xray_traces:
             xray_context = _get_xray_trace_context()
             if xray_context is not None:
-                context["parent-id"] = xray_context.get("parent-id")
-
-        headers = _context_obj_to_headers(context)
-        span_context = propagator.extract(headers)
-        tracer.context_provider.activate(span_context)
+                dd_trace_context.span_id = xray_context.span_id
+                dd_trace_context.trace_id = xray_context.trace_id
+                dd_trace_context.sampling_priority = xray_context.sampling_priority
+                dd_trace_context.dd_origin = xray_context.dd_origin
```

Comment:
Previously we were only setting `parent-id` from xray context. We shouldn't change that functionality.